sympy/integrals/integrals.py

```diff
@@ -1404,6 +1406,79 @@ def integrate(*args, **kwargs):
     else:
         return integral
 
+def cauchy_principal_value(func, sym, lower_limit = -oo, upper_limit = oo):
+    """
+    Compute the Cauchy Principle Value of the definite integral of a real function on the entire real axis.
+    In mathematics, the Cauchy principal value, is a method for assigning values to certain improper
+    integrals which would otherwise be undefined.
+    Cauchy Principle Value is  defined as:
+          +oo               +r
+          /                  /
+         /                  /
+         |                  |
+         |                  |
+    p.v. | f(x)dx  = lim    | f(x)dx
+         |          r -> oo |
+         |                  |
+         /                  /
+        /                  /
+     -oo               -r
+
+    Examples
+    ========
+
+    >>> from sympy import Dummy, symbols, integrate, limit, oo
+    >>> from sympy.integrals.integrals import cauchy_principal_value
+    >>> from sympy.calculus.singularities import singularities
+    >>> x = symbols('x', real = True)
+    >>> g = x + 1
+    >>> cauchy_principal_value(g,x)
+    oo
+    >>> f = 1/(x**3)
+    >>> singularities(f,x)
+    {0}
+    >>> cauchy_principal_value(g,x)
+    0
+    >>> cauchy_principal_value(g,x, -oo, 0)
+    -oo
+    >>> cauchy_principal_value(g,x, 0, oo)
+    oo
+
+
+    See also
+    ========
+
+    [1] https://en.wikipedia.org/wiki/Cauchy_principal_value
+    [2] http://mathworld.wolfram.com/CauchyPrincipalValue.html
+    """
+
+    r = Dummy('r')
+    if (sym.is_real == False) or (func.is_real == False):
+        raise ValueError("The function should be real valued for calculation of Cauchy Principle Value")
+    else:
+        if lower_limit == -oo and upper_limit == oo :
```

Comment:
There can be singularities also in this case. They should be taken into account.

Reply:
I'll correct this @jksuom . Thanks for raising this point.

Reply:
@jksuom  The problem is coming when I'm performing this ` -oo + oo ` as this is giving `nan` even using basic `integrate` ( when I took a monotonic function `f = x + 1` and performed integration from `-oo to oo` ). So whenever I'm adding the broken down integrals, if say from `-oo to 0` and `0 to +oo` then it eventually results in ` -oo + oo ` which it is unable to solve and results in `nan` . So what do you recommend me to do? We could also mention in the documentation that in such a case where we expect `0`, there it results in `nan`.